src/files-and-videos/videos/data/api.js

```diff
@@ -0,0 +1,238 @@
+/* eslint-disable import/prefer-default-export */
+import { camelCaseObject, ensureConfig, getConfig } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+import saveAs from 'file-saver';
+import { isEmpty } from 'lodash';
+
+ensureConfig([
+  'STUDIO_BASE_URL',
+], 'Course Apps API service');
+
+export const getApiBaseUrl = () => getConfig().STUDIO_BASE_URL;
+export const getVideosUrl = (courseId) => `${getApiBaseUrl()}/api/contentstore/v1/videos/${courseId}`;
+export const getCoursVideosApiUrl = (courseId) => `${getApiBaseUrl()}/videos/${courseId}`;
+
+/**
+ * Fetches the course custom pages for provided course
+ * @param {string} courseId
+ * @returns {Promise<[{}]>}
+ */
+export async function getVideos(courseId) {
+  const { data } = await getAuthenticatedHttpClient()
+    .get(getVideosUrl(courseId));
+  const { video_transcript_settings: videoTranscriptSettings } = data;
+  const { transcription_plans: transcriptionPlans } = videoTranscriptSettings;
+  return {
+    ...camelCaseObject(data),
+    videoTranscriptSettings: {
+      ...camelCaseObject(videoTranscriptSettings),
+      transcriptionPlans,
+    },
+  };
+}
+
+/**
+ * Fetches the course custom pages for provided course
+ * @param {string} courseId
+ * @returns {Promise<[{}]>}
+ */
+export async function fetchVideoList(courseId) {
+  const { data } = await getAuthenticatedHttpClient()
+    .get(getCoursVideosApiUrl(courseId));
+  return camelCaseObject(data);
+}
+
+export async function deleteTranscript({ videoId, language, apiUrl }) {
+  await getAuthenticatedHttpClient()
+    .delete(`${getApiBaseUrl()}${apiUrl}/${videoId}/${language}`);
+}
+
+export async function downloadTranscript({
+  videoId,
+  language,
+  apiUrl,
+  filename,
+}) {
+  const { data } = await getAuthenticatedHttpClient()
+    .get(`${getApiBaseUrl()}${apiUrl}?edx_video_id=${videoId}&language_code=${language}`);
+  const file = new Blob([data], { type: 'text/plain;charset=utf-8' });
+  saveAs(file, filename);
+}
+
+export async function uploadTranscript({
+  videoId,
+  newLanguage,
+  apiUrl,
+  file,
+  language,
+}) {
+  const formData = new FormData();
+  formData.append('file', file);
+  formData.append('edx_video_id', videoId);
+  formData.append('language_code', language);
+  formData.append('new_language_code', newLanguage);
+  await getAuthenticatedHttpClient().post(`${getApiBaseUrl()}${apiUrl}`, formData);
+}
+
+export async function getDownload(selectedRows) {
```

Comment:
NICE. Just saying that I saw so many even senior devs get this handling of multiple promises wrong...